### ISSUE TITLE
PHP syntax error on multiple line string concatenation in a property assignment of the file lang/multilang.php

### ISSUE THREAD

**Issue by leonardodeconti** [maths/moodle-qtype_stack#458]
Hello Dears,

First, thanks you guys for your efforts to do this amazing plugin!

Well, after I did an update of the STACK plugin to the version "4.2.1 for Moodle 3.0+" I got the following error when I load the page "Question bank" (https://URL/question/edit.php?courseid=1):

> PHP Parse error:  syntax error, unexpected '.', expecting ',' or ';' in /pathto/moodle/question/type/stack/lang/multilang.php on line 37, referer: https://URL/question/edit.php?courseid=1

At line 37 of multilang.php file we have the following string concatenation divided in two lines in a property assignment:
```
private $search = '/(<span(\s+lang="[a-zA-Z0-9_-]+"|\s+class="multilang"){2}\s*>.*?<\/span>)'
			. '(\s*<span(\s+lang="[a-zA-Z0-9_-]+"|\s+class="multilang"){2}\s*>.*?<\/span>)+/is';

```
So I have removed the string concatenation making a "straight assignment":
`private $search = '/(<span(\s+lang="[a-zA-Z0-9_-]+"|\s+class="multilang"){2}\s*>.*?<\/span>)(\s*<span(\s+lang="[a-zA-Z0-9_-]+"|\s+class="multilang"){2}\s*>.*?<\/span>)+/is';`
	
and the error gone out!

I suspect the error occurs because I have an old version of PHP:

> $ php --version
> PHP 5.5.9-1ubuntu4.26 (cli) (built: Sep 17 2018 13:46:30) 

But my Moodle version is 3.1.17 that requires PHP 5.4.4+ ... so I think I'm not incompatible with [Moodle Server requirements](https://docs.moodle.org/dev/Moodle_3.1_release_notes#Server_requirements).

Maybe we have a bug because of incompatibilities between PHP versions ?

**Comment by aharjula** [comment]
Well, that is interesting. It sure looks good but maybe there is some magic character there that messes things, or maybe it is just the old PHP as you say. 

The bad news for you is that the requirements of STACK will probably soon change so that old of a Moodle version will not be supported, however I am not quite certain whether we jump all the way to Moodle 3.5 (or 3.4, probably 3.5 due to LTS) and thus PHP 7 but I am pretty certain 3.1 is too old. It could be that we make that change (by adding new features that require newer PHP) around the same time as the Moodle long time support release 3.1 reaches end of life but we might do it earlier and just provide a link to older STACK if one needs to use older Moodle.

https://docs.moodle.org/dev/Releases#Version_support

**Comment by sangwinc** [comment]
Well, that is strange, and possibly embarrassing!

I honestly have no idea what is wrong with that.  I normally put the concatenation dot at the end of the previous line, e.g. see https://github.com/maths/moodle-qtype_stack/commit/fa59892819b0912c1c8f1330c7f1400f728e37b6) but is it really that?

I don't think we will be testing as far back as Moodle 3.1 for much longer.  Sorry.

**Comment by leonardodeconti** [comment]
> Well, that is strange, and possibly embarrassing!
> 
> I honestly have no idea what is wrong with that. I normally put the concatenation dot at the end of the previous line, e.g. see [fa59892](https://github.com/maths/moodle-qtype_stack/commit/fa59892819b0912c1c8f1330c7f1400f728e37b6)) but is it really that?
> 
> I don't think we will be testing as far back as Moodle 3.1 for much longer. Sorry.

Hello sangwinc.

Usually the position of concatenation dot doesn't matter. It may be at the end of first line or on the next one. Don't worry, you didn't do this mistake. =]

I searched a little more and found the PHP RFC that brings static scalar expressions (like concatenation) in the Class Properties initilization. It is available only in PHP versions >= 5.6: https://wiki.php.net/rfc/const_scalar_exprs.#class_property_declarations


Before that version we can only use literal (like strings, integers and so on) or constant values on Class Properties initilization.
So because of this incompatibility I have got this PHP parser error.

I had opened this issue just because (in strict terms of Moodle Server requirements) this implementation (that uses PHP 5.5) should be enough for support Moodle 3.1 and its plugins. Don't you agree?
And I tought that others users can be affected by this incompatibility.

But don't stress with backward compatibility, we use open source so we can apply our customizations.
And, the most important: if the server becomes fully incompatible with newer Moodle requirements, I can sell also a job for server upgrade in addition to the simple moodle update! hahahahaha

**Comment by sangwinc** [comment]
Thank you very much for explaining this.  It is probably not going to affect lots of people, but we should know about it.  PHP 5.6 is also out of support, and it looks like PHP 5.5 support stopped in 2016 (https://en.wikipedia.org/wiki/PHP#PHP_5).  So, I think it is reasonable to require PHP at least 5.6 from this point on for STACK.  I'll make that clear in the docs....

Chris
